torch/csrc/autograd/python_variable.cpp

```diff
@@ -231,6 +231,22 @@ int THPVariable_set_data(THPVariable *self, PyObject *data)
   END_HANDLE_TH_ERRORS_RET(-1)
 }
 
+static PyObject * THPVariable_get_int_repr(THPVariable *self)
+{
+  HANDLE_TH_ERRORS
+  /// NOTE: Previously, if we change the tensor metadata (e.g. sizes / strides /
+  /// storage / storage_offset) of a tensor created from `.data`, those metadata
+  /// in the original tensor will also be updated. However, the new behavior is that
+  /// those metadata changes to the `.data` tensor will not update the original tensor
+  /// anymore, and here we need to set `allow_tensor_metadata_change_` to false to
+  /// make such changes explicitly illegal, in order to prevent users from changing
+  /// metadata of the `.data` tensor and expecting the original tensor to also
+  /// be updated.
+  auto& self_ = self->cdata;
+  return torch::autograd::utils::wrap(self_.int_repr());
```

Comment:
wait, which overload of wrap is this calling?  When I try `type(...)` on the test I see a numpy array?

Reply:
hmm, this is probably not tested, I think the change in tools/autograd/templates/python_variable_methods.cpp is the one that takes effect.